### PR TITLE
Refactor Event & Schedule Filters

### PR DESCRIPTION
**Pull request by diogogmatos**
Refactored the code for Event & Schedule Filters, which was suffering from a huge lack of modularity and was repeating a lot of code, adding a level of complexity and size that made it hard to understand the code and make changes or improvements.

For this:
- Created new `FilterBlock` component that is responsible for creating the list of filters for both Events and Schedule pages. It reuses the code for both of them, applying all the same styling and adapting to specific needs in accordance with the variables that it receives.
- Created a universal `CheckBoxProps` type that allows to pass information about the filters into `FilterBlock` in a normalized manner.
-  Documented the code for better understanding of each process.
- Converted the styling of the filters to TailwindCSS and removed the now unnecessary styling modules.
- Removed the "5th year" collapse from the Schedule page, since schedules don't apply to 5th year.

This resulted in a much more readable code, that preserves the same functionality whilst reducing the size of the code.

### CODE REVIEW

File: components/FilterBlock/FilterBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+import { Fragment, useState, useEffect } from "react";
+
+import { CheckBoxProps, SelectedShift } from "../../types";
+
+type FilterBlockProps = {
+  layer1: string[]; // contains the titles for the collapse's of the 1st layer
+  layer2?: string[]; // contains the titles for the collapse's of the 2nd layer
+  checkBoxes: CheckBoxProps[][]; // contains the checkboxes information in a universal format
+  exception?: number; // indicates the index of an element from layer1 where the layer2 should be ignored, for example "5th year"
+  checked: any; // assumes different types when called from ScheduleFilters.tsx and EventFilters.tsx
```
Comment (RuiL1904) [inline components/FilterBlock/FilterBlock.tsx:13]: You could create a union of types, then.

File: components/FilterBlock/FilterBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,360 @@
+import { Checkbox, Collapse } from "antd";
+import "antd/dist/reset.css";
+
+import { Fragment, useState, useEffect } from "react";
+
+import { CheckBoxProps, SelectedShift } from "../../types";
+
+type FilterBlockProps = {
+  layer1: string[]; // contains the titles for the collapse's of the 1st layer
```
Comment (RuiL1904) [inline components/FilterBlock/FilterBlock.tsx:9]:
```suggestion
  layer1: string[]; // contains the titles for the collapses of the 1st layer
```

File: components/FilterBlock/FilterBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+  layer2?: string[]; // contains the titles for the collapse's of the 2nd layer
```
Comment (RuiL1904) [inline components/FilterBlock/FilterBlock.tsx:10]:
```suggestion
  layer2?: string[]; // contains the titles for the collapses of the 2nd layer
```

File: components/FilterBlock/FilterBlock.tsx (hunk 1 above)
Comment (diogogmatos) [reply components/FilterBlock/FilterBlock.tsx:13]: Done.

**Review by RuiL1904 — APPROVED** [review]

**Pull request merged**